### PR TITLE
added llm-lab-frontend deployment

### PR DESCRIPTION
**Pull request by daniel-jeremia**

### CODE REVIEW

File: charts/datasaur/values.yaml [hunk 1]
```diff
@@ -155,6 +172,8 @@ imageCredentials:
 
 ingress:
   host: null
+  llm:
```
Comment (tonoman3g) [inline charts/datasaur/values.yaml]: maybe we could use terminology llmLabs across the helm chart

so this should be llmLabs.host

File: charts/datasaur/values.yaml [hunk 2]
```diff
@@ -116,6 +116,15 @@ services:
       limits:
         cpu: null
         memory: null
+  llmLabFrontend:
```
Comment (tonoman3g) [inline charts/datasaur/values.yaml]: llmLabsFrontend

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{{- if .Values.ingress.llm.host }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-llm-lab-frontend-service
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: llm-labs-***

File: charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{{- if not .Values.llmTls.secretName }}
+apiVersion: v1
```
Comment (tonoman3g) [inline charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml]: the file name should be datasaur-llm-labs-tls.yaml

File: charts/datasaur/values.yaml [hunk 5]
```diff
@@ -145,6 +154,14 @@ tls:
   errorPage: null
   secretName: null
 
+llmTls:
```
Comment (tonoman3g) [inline charts/datasaur/values.yaml]: this is also should be llmLabsTls

File: charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  name: {{ .Release.Name }}-datasaur-llm-lab-tls
+type: kubernetes.io/tls
+{{- if .Values.llmTls.ca }}
+---
+apiVersion: v1
+data:
+  ca.crt: {{ .Values.llmTls.ca }}
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-datasaur-llm-lab-ca
```
Comment (tonoman3g) [inline charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml]: -datasaur-llm-labs-ca

File: charts/datasaur/templates/main/ingress.yaml [hunk 7]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.ingress.host }}
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/ingress.yaml:1]: by adding if here, it means that we want host variable to be optional, is it expected?

File: charts/datasaur/templates/main/llm-ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            backend:
+              service:
+                name: {{ .Release.Name }}-backend-service
+                port:
+                  number: 8080
+          - path: /(.*)
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ .Release.Name }}-llm-lab-frontend-service
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-ingress.yaml]: llm-labs-frontend-service

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    metadata:
+      annotations:
+        rollme: {{ randAlphaNum 5 | quote }}
+      labels:
+        app: {{ .Release.Name }}-llm-lab-frontend
+    spec:
+{{ include "imagePullSecretTemplate" . | nindent 6 }}
+      containers:
+{{- if .Values.image.internal }}
+        - image: {{ .Values.image.registry | trimSuffix "/" }}{{ if .Values.image.registry }}/{{ end }}vulcan-ai/vulcan/llm-lab-frontend:{{ .Values.image.dockerTag }}
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: hmm, unfortunately the image name is llm-lab-frontend ya :(

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+              mountPath: /usr/src/app/.env
+              subPath: .env
+          envFrom:
+            - configMapRef:
+                name: datasaur-env
+            - secretRef:
+                name: datasaur-secret
+          env:
+            - name: ASSETS_URL
+              value: https://$(HOST)
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: for asset, why do we use $HOST here? $HOST is the NLP host right?

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+              cpu: {{ .Values.services.llmLabFrontend.resources.requests.cpu }}
+              memory: {{ .Values.services.llmLabFrontend.resources.requests.memory }}
+          {{- end }}
+      volumes:
+        - name: frontend-env
+          configMap:
+            name: frontend-env
+            items:
+              - key: .env
+                path: .env
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: are you reusing the env var for NLP frontend?

File: charts/datasaur/templates/main/ingress.yaml (hunk 7 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/main/ingress.yaml:1]: yes, if they only want to install llm labs, then ingress.host is not required

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml (hunk 10 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: hmm, maybe I should make like LLM_HOST as well?

Comment (tonoman3g) [reply charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: yes

File: charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
+{{- if not .Values.llmTls.secretName }}
+apiVersion: v1
+data:
+  tls.crt: {{ .Values.llmTls.cert }}
```
Comment (tonoman3g) [inline charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml]: oh ya, any minimum datasaur version that contains LLM Labs? @daniel-jeremia

File: charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml (hunk 4 above)
Comment (tonoman3g) [reply charts/datasaur/templates/config/datasaur-llm-lab-tls.yaml]: bump, this the file name still not changed.


additionally please add the minimum datasaur version requirement @daniel-jeremia

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,70 @@
+{{- if .Values.ingress.llmLabs.host }}
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: this file also should be renamed to llm-labs-frontend-deployment.yaml

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml (hunk 10 above)
Comment (tonoman3g) [reply charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: how is it? bumping this

File: charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml (hunk 13 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/main/llm-lab-frontend-deployment.yaml]: done @tonoman3g

File: charts/datasaur/Chart.yaml [hunk 14]
```diff
@@ -15,13 +15,13 @@ type: application
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 2.0.2
+version: 2.1.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: ">=6.39.0"
+appVersion: ">=6.44.0"
```
Comment (tonoman3g) [inline charts/datasaur/Chart.yaml]: should we put 6.46 here? @daniel-jeremia

File: charts/datasaur/templates/main/llm-labs-frontend-deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+              mountPath: /usr/src/app/.env
+              subPath: .env
+          envFrom:
+            - configMapRef:
+                name: datasaur-env
+            - secretRef:
+                name: datasaur-secret
+          env:
+            - name: ASSETS_URL
+              value: https://$(LLM_LABS_HOST)
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/llm-labs-frontend-deployment.yaml:53]: sorry, who does set this value?

File: charts/datasaur/Chart.yaml (hunk 14 above)
Comment (daniel-jeremia) [reply charts/datasaur/Chart.yaml]: hmmm, yea i guess that makes sense @tonoman3g

File: charts/datasaur/templates/main/llm-labs-frontend-deployment.yaml (hunk 15 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/main/llm-labs-frontend-deployment.yaml:53]: the user, I already included it in our values.sample.yaml @tonoman3g

**Review by tonoman3g — APPROVED** [review]

**Pull request merged**